
project packages/apps/Snap/
diff --git a/packages/apps/Snap/src/com/android/camera/VideoModule.java b/packages/apps/Snap/src/com/android/camera/VideoModule.java
index fd7bff6..b1c543b 100644
--- a/packages/apps/Snap/src/com/android/camera/VideoModule.java
+++ b/packages/apps/Snap/src/com/android/camera/VideoModule.java
@@ -561,7 +561,7 @@ public class VideoModule implements CameraModule,
 
     private boolean takeASnapshot() {
         // Only take snapshots if video snapshot is supported by device
-        if (CameraUtil.isVideoSnapshotSupported(mParameters) && !mIsVideoCaptureIntent) {
+        if (!CameraUtil.isVideoSnapshotSupported(mParameters) && !mIsVideoCaptureIntent) {
             if (!mMediaRecorderRecording || mPaused || mSnapshotInProgress) {
                 return false;
             }
