
project packages/apps/Snap/
diff --git a/packages/apps/Snap/src/com/android/camera/VideoModule.java b/packages/apps/Snap/src/com/android/camera/VideoModule.java
index fd7bff6..a69a22c 100644
--- a/packages/apps/Snap/src/com/android/camera/VideoModule.java
+++ b/packages/apps/Snap/src/com/android/camera/VideoModule.java
@@ -2883,7 +2883,6 @@ public class VideoModule implements CameraModule,
     }
 
     private void showTapToSnapshotToast() {
-        new RotateTextToast(mActivity, R.string.video_snapshot_hint, 0).show();
         // Clear the preference.
         Editor editor = mPreferences.edit();
         editor.putBoolean(CameraSettings.KEY_VIDEO_FIRST_USE_HINT_SHOWN, false);
