
project frameworks/av/
diff --git a/frameworks/av/media/libstagefright/ACodec.cpp b/frameworks/av/media/libstagefright/ACodec.cpp
index 9968ee0..7d9ff5f 100644
--- a/frameworks/av/media/libstagefright/ACodec.cpp
+++ b/frameworks/av/media/libstagefright/ACodec.cpp
@@ -98,6 +98,10 @@
 
 #include "include/avc_utils.h"
 
+#if defined(OMAP_ENHANCEMENT) || defined(OMAP_TUNA)
+#include <cutils/properties.h>
+#endif
+
 #ifdef ENABLE_AV_ENHANCEMENTS
 #include <QCMediaDefs.h>
 #include <ExtendedUtils.h>
@@ -3172,7 +3176,8 @@ status_t ACodec::setupAVCEncoderParameters(const sp<AMessage> &msg) {
         if (err != OK) {
             ALOGE("Setting intra macroblock refresh mode (%d) failed: 0x%x",
                     err, intraRefreshMode);
-            return err;
+            ALOGE("setupAVCEncoderParameters() SKIP!! intra-refresh-mode");
+            //return err;
         }
     }
 
@@ -4980,6 +4985,14 @@ bool ACodec::BaseState::onOMXFillBufferDone(
 
             reply->setInt32("buffer-id", info->mBufferID);
 
+#if defined(OMAP_ENHANCEMENT) || defined(OMAP_TUNA)
+            char value[PROPERTY_VALUE_MAX];
+            if (property_get("persist.gallery.player", value, "0")) {
+                if (!strcmp(value, "1")) {
+                    reply->setInt32("render", true);
+                }
+            }
+#endif
             notify->setMessage("reply", reply);
 
             notify->post();
