
project frameworks/av/
diff --git a/frameworks/av/media/libstagefright/ACodec.cpp b/frameworks/av/media/libstagefright/ACodec.cpp
index a110eae..c2010c5 100644
--- a/frameworks/av/media/libstagefright/ACodec.cpp
+++ b/frameworks/av/media/libstagefright/ACodec.cpp
@@ -3166,7 +3170,8 @@ status_t ACodec::setupAVCEncoderParameters(const sp<AMessage> &msg) {
         if (err != OK) {
             ALOGE("Setting intra macroblock refresh mode (%d) failed: 0x%x",
                     err, intraRefreshMode);
-            return err;
+            ALOGE("setupAVCEncoderParameters() SKIP!! intra-refresh-mode");
+            //return err;
         }
     }
 
