
project vendor/cm/
diff --git a/vendor/cm/config/common_full.mk b/vendor/cm/config/common_full.mk
index 5d12550..9d1e4e3 100644
--- a/vendor/cm/config/common_full.mk
+++ b/vendor/cm/config/common_full.mk
@@ -6,21 +6,10 @@ include vendor/cm/config/cm_audio.mk
 
 # Optional CM packages
 PRODUCT_PACKAGES += \
-    Galaxy4 \
-    HoloSpiralWallpaper \
-    LiveWallpapers \
-    LiveWallpapersPicker \
-    MagicSmokeWallpapers \
-    NoiseField \
-    PhaseBeam \
-    VisualizationWallpapers \
-    PhotoTable \
-    SoundRecorder \
-    PhotoPhase
+    SoundRecorder
 
 # Extra tools in CM
 PRODUCT_PACKAGES += \
-    vim \
     zip \
     unrar \
     curl
