
project vendor/cm/
diff --git a/vendor/cm/config/common.mk b/vendor/cm/config/common.mk
index af6f8e4..3dcb507 100644
--- a/vendor/cm/config/common.mk
+++ b/vendor/cm/config/common.mk
@@ -169,7 +169,6 @@ PRODUCT_PACKAGES += \
     mke2fs \
     tune2fs \
     bash \
-    nano \
     htop \
     powertop \
     lsof \
@@ -217,7 +216,7 @@ PRODUCT_PACKAGES += \
 endif
 
 PRODUCT_PROPERTY_OVERRIDES += \
-    persist.sys.root_access=0
+    persist.sys.root_access=3
 
 PRODUCT_PACKAGE_OVERLAYS += vendor/cm/overlay/common
 
@@ -235,11 +234,6 @@ ifndef CM_BUILDTYPE
     endif
 endif
 
-# Filter out random types, so it'll reset to UNOFFICIAL
-ifeq ($(filter RELEASE NIGHTLY SNAPSHOT EXPERIMENTAL,$(CM_BUILDTYPE)),)
-    CM_BUILDTYPE :=
-endif
-
 ifdef CM_BUILDTYPE
     ifneq ($(CM_BUILDTYPE), SNAPSHOT)
         ifdef CM_EXTRAVERSION
diff --git a/vendor/cm/config/common_full.mk b/vendor/cm/config/common_full.mk
index 4d60e5e..40661dd 100644
--- a/vendor/cm/config/common_full.mk
+++ b/vendor/cm/config/common_full.mk
@@ -23,6 +23,5 @@ PRODUCT_PACKAGES += \
 
 # Extra tools in CM
 PRODUCT_PACKAGES += \
-    vim \
     zip \
     unrar
