
project frameworks/av/
diff --git a/frameworks/av/media/libstagefright/id3/ID3.cpp b/frameworks/av/media/libstagefright/id3/ID3.cpp
index cf27be3..f00bb89 100644
--- a/frameworks/av/media/libstagefright/id3/ID3.cpp
+++ b/frameworks/av/media/libstagefright/id3/ID3.cpp
@@ -483,7 +483,7 @@ void ID3::Iterator::getString(String8 *id, String8 *comment) const {
 // comment fields (COM/COMM) contain an initial short descriptor, followed by \0,
 // followed by more data. The data following the \0 can be retrieved by setting
 // "otherdata" to true.
-void ID3::Iterator::getstring(String8 *id, bool otherdata) const {
+void __attribute__((optimize("no-tree-vectorize"))) ID3::Iterator::getstring(String8 *id, bool otherdata) const {
     id->setTo("");
 
     size_t size;
@@ -637,6 +637,8 @@ void ID3::Iterator::findFrame() {
                 | (mParent.mData[mOffset + 4] << 8)
                 | mParent.mData[mOffset + 5];
 
+            if( mFrameSize == 0 ) return;
+
             mFrameSize += 6;
 
             // Prevent integer overflow in validation
