
project bootable/recovery/
diff --git a/bootable/recovery/voldclient/Android.mk b/bootable/recovery/voldclient/Android.mk
index 0c3f505..908656a 100644
--- a/bootable/recovery/voldclient/Android.mk
+++ b/bootable/recovery/voldclient/Android.mk
@@ -3,7 +3,7 @@ LOCAL_PATH := $(call my-dir)
 include $(CLEAR_VARS)
 LOCAL_MODULE := libvoldclient
 LOCAL_SRC_FILES := commands.c dispatcher.c event_loop.c
-LOCAL_CFLAGS := -DMINIVOLD -Werror
+LOCAL_CFLAGS := -DMINIVOLD -Werror -Wno-error=unused-parameter
 LOCAL_C_INCLUDES :=         	\
     $(LOCAL_PATH)/..       	\
     system/core/fs_mgr/include	\
