
project packages/services/Telephony/
diff --git a/packages/services/Telephony/src/com/android/phone/PhoneGlobals.java b/packages/services/Telephony/src/com/android/phone/PhoneGlobals.java
index de1ecd5..98f8fe7 100644
--- a/packages/services/Telephony/src/com/android/phone/PhoneGlobals.java
+++ b/packages/services/Telephony/src/com/android/phone/PhoneGlobals.java
@@ -87,6 +87,10 @@ import static com.android.internal.telephony.MSimConstants.SUBSCRIPTION_KEY;
 
 import org.codeaurora.ims.csvt.ICsvtService;
 
+// Patch airplane
+import android.net.ConnectivityManager;
+//End Patch
+
 /**
  * Global state for the telephony subsystem when running in the primary
  * phone process.
@@ -1188,7 +1192,21 @@ public class PhoneGlobals extends ContextWrapper implements WiredHeadsetListener
                 // enabled here implies airplane mode is OFF from above condition
                 SystemProperties.set(PROPERTY_AIRPLANE_MODE_ON, (enabled ? "0" : "1"));
 
+                // Patch. Airplane
+                ConnectivityManager cm = (ConnectivityManager)getSystemService(Context.CONNECTIVITY_SERVICE);
+                // End patch
+
                 phone.setRadioPower(enabled);
+
+                // Patch. Airplane
+                if (enabled) {
+                    boolean dmode = cm.getMobileDataEnabled();
+                    cm.setMobileDataEnabled(false);
+                    cm.setMobileDataEnabled(true);
+                    cm.setMobileDataEnabled(dmode);
+                }
+                // End patch
+
             } else if (action.equals(TelephonyIntents.ACTION_ANY_DATA_CONNECTION_STATE_CHANGED)) {
                 if (VDBG) Log.d(LOG_TAG, "mReceiver: ACTION_ANY_DATA_CONNECTION_STATE_CHANGED");
                 if (VDBG) Log.d(LOG_TAG, "- state: " + intent.getStringExtra(PhoneConstants.STATE_KEY));
