
project system/core/
diff --git a/system/core/rootdir/init.rc b/system/core/rootdir/init.rc
index dbbf512..947d440 100644
--- a/system/core/rootdir/init.rc
+++ b/system/core/rootdir/init.rc
@@ -462,8 +462,8 @@ service console /system/bin/sh
     user shell
     group log
 
-service auditd /system/bin/auditd -k
-    class main
+#service auditd /system/bin/auditd -k
+#    class main
 
 on property:ro.debuggable=1
     start console
