
project system/vold/
diff --git a/system/vold/Android.mk b/system/vold/Android.mk
index 9020ba3..7be2e03 100644
--- a/system/vold/Android.mk
+++ b/system/vold/Android.mk
@@ -72,7 +72,8 @@ common_libraries := \
 	liblog \
 	libdiskconfig \
 	libext2_blkid \
-	liblogwrap
+	liblogwrap \
+	libselinux
 
 common_static_libraries := \
 	libfs_mgr \
diff --git a/system/vold/VolumeManager.cpp b/system/vold/VolumeManager.cpp
index b288b70..003e59a 100644
--- a/system/vold/VolumeManager.cpp
+++ b/system/vold/VolumeManager.cpp
@@ -36,6 +36,8 @@
 #include <cutils/log.h>
 #include <cutils/properties.h>

+#include <selinux/android.h>
+
 #include <sysutils/NetlinkEvent.h>

 #include <private/android_filesystem_config.h>
@@ -655,6 +657,12 @@ int VolumeManager::fixupAsecPermissions(const char *id, gid_t gid, const char* f
             } else if (ftsent->fts_info & FTS_F) {
                 result |= fchmod(fd, privateFile ? 0640 : 0644);
             }
+
+           if (selinux_android_restorecon(ftsent->fts_path) < 0) {
+                SLOGE("restorecon failed for %s: %s\n", ftsent->fts_path, strerror(errno));
+                result |= -1;
+            }
+
             close(fd);
         }
         fts_close(fts);
