
project packages/apps/Settings/
diff --git a/packages/apps/Settings/src/com/android/settings/wifi/AdvancedWifiSettings.java b/packages/apps/Settings/src/com/android/settings/wifi/AdvancedWifiSettings.java
index bafb515..6eca087 100644
--- a/packages/apps/Settings/src/com/android/settings/wifi/AdvancedWifiSettings.java
+++ b/packages/apps/Settings/src/com/android/settings/wifi/AdvancedWifiSettings.java
@@ -227,6 +227,9 @@ public class AdvancedWifiSettings extends SettingsPreferenceFragment
 
         if (KEY_COUNTRY_CODE.equals(key)) {
             try {
+		Settings.Global.putString(getContentResolver(),
+                        Settings.Global.WIFI_COUNTRY_CODE,
+                        (String) newValue);
                 mWifiManager.setCountryCode((String) newValue, true);
             } catch (IllegalArgumentException e) {
                 Toast.makeText(getActivity(), R.string.wifi_setting_countrycode_error,
