
project frameworks/base/
diff --git a/frameworks/base/packages/SettingsProvider/res/values/defaults.xml b/frameworks/base/packages/SettingsProvider/res/values/defaults.xml
index 7284843..dc4ddfd 100644
--- a/frameworks/base/packages/SettingsProvider/res/values/defaults.xml
+++ b/frameworks/base/packages/SettingsProvider/res/values/defaults.xml
@@ -222,4 +222,9 @@
 
     <!-- Default value of Settings.System.QUICK_SETTINGS_RIBBON_TILES -->
     <string name="def_quick_settings_ribbon_tiles"></string>
+
+    <!-- Default for wifi country code: US(1~11)EU(1~13)JP(1~14) -->
+    <add-resource name="wifi_country_code" type="string" />
+    <string name="wifi_country_code">JP</string>
+
 </resources>
diff --git a/frameworks/base/packages/SettingsProvider/src/com/android/providers/settings/DatabaseHelper.java b/frameworks/base/packages/SettingsProvider/src/com/android/providers/settings/DatabaseHelper.java
index 4d1fbfb..3171ec5 100644
--- a/frameworks/base/packages/SettingsProvider/src/com/android/providers/settings/DatabaseHelper.java
+++ b/frameworks/base/packages/SettingsProvider/src/com/android/providers/settings/DatabaseHelper.java
@@ -2349,6 +2349,8 @@ public class DatabaseHelper extends SQLiteOpenHelper {
             loadStringSetting(stmt, Settings.Global.AIRPLANE_MODE_TOGGLEABLE_RADIOS,
                     R.string.airplane_mode_toggleable_radios);
 
+	    loadStringSetting(stmt, Settings.Global.WIFI_COUNTRY_CODE, R.string.wifi_country_code);
+
             loadBooleanSetting(stmt, Settings.Global.ASSISTED_GPS_ENABLED,
                     R.bool.assisted_gps_enabled);
 
