
project system/core/
diff --git a/system/core/rootdir/init.rc b/system/core/rootdir/init.rc
index 9ae2f94..4d834f0 100644
--- a/system/core/rootdir/init.rc
+++ b/system/core/rootdir/init.rc
@@ -464,8 +464,8 @@ service console /system/bin/sh
     group log
     seclabel u:r:shell:s0
 
-service auditd /system/bin/auditd -k
-    class main
+#service auditd /system/bin/auditd -k
+#    class main
 
 on property:ro.debuggable=1
     start console
