
project system/core/
diff --git a/system/core/rootdir/init.rc b/system/core/rootdir/init.rc
index 61f41c8..75745a2 100644
--- a/system/core/rootdir/init.rc
+++ b/system/core/rootdir/init.rc
@@ -478,8 +478,8 @@ service console /system/bin/sh
     group log
     seclabel u:r:shell:s0
 
-service auditd /system/bin/auditd -k
-    class main
+#service auditd /system/bin/auditd -k
+#    class main
 
 on property:ro.debuggable=1
     start console
