
project system/vold/
diff --git a/system/vold/Android.mk b/system/vold/Android.mk
index 7037292..31f3542 100644
--- a/system/vold/Android.mk
+++ b/system/vold/Android.mk
@@ -20,7 +20,7 @@ ifeq ($(BOARD_VOLD_CRYPTFS_MIGRATE), true)
 common_cflags += -DCRYPTFS_MIGRATE
 endif
 
-common_cflags += -Werror
+common_cflags += -Werror -Wno-error=unused-parameter
 
 common_src_files := \
 	VolumeManager.cpp \
