
project vendor/cm/
diff --git a/vendor/cm/config/common.mk b/vendor/cm/config/common.mk
index b83b994..6d03800 100644
--- a/vendor/cm/config/common.mk
+++ b/vendor/cm/config/common.mk
@@ -141,7 +141,6 @@ PRODUCT_PACKAGES += \
     audio_effects.conf \
     CMWallpapers \
     Apollo \
-    CMFileManager \
     LockClock \
     CMUpdater \
     CMFota \
@@ -232,9 +231,9 @@ ifndef CM_BUILDTYPE
 endif
 
 # Filter out random types, so it'll reset to UNOFFICIAL
-ifeq ($(filter RELEASE NIGHTLY SNAPSHOT EXPERIMENTAL,$(CM_BUILDTYPE)),)
-    CM_BUILDTYPE :=
-endif
+#ifeq ($(filter RELEASE NIGHTLY SNAPSHOT EXPERIMENTAL,$(CM_BUILDTYPE)),)
+#    CM_BUILDTYPE :=
+#endif
 
 ifdef CM_BUILDTYPE
     ifneq ($(CM_BUILDTYPE), SNAPSHOT)
diff --git a/vendor/cm/config/common_full.mk b/vendor/cm/config/common_full.mk
index 2eccc56..77f9206 100644
--- a/vendor/cm/config/common_full.mk
+++ b/vendor/cm/config/common_full.mk
@@ -2,7 +2,7 @@
 $(call inherit-product, vendor/cm/config/common.mk)
 
 # Bring in all video files
-$(call inherit-product, frameworks/base/data/videos/VideoPackage2.mk)
+#$(call inherit-product, frameworks/base/data/videos/VideoPackage2.mk)
 
 # Include CM audio files
 include vendor/cm/config/cm_audio.mk
@@ -33,5 +33,5 @@ PRODUCT_PACKAGES += \
     libvideoeditorplayer
 
 # Extra tools in CM
-PRODUCT_PACKAGES += \
-    vim
+#PRODUCT_PACKAGES += \
+#    vim
