
project frameworks/opt/telephony/
diff --git a/frameworks/opt/telephony/src/java/com/android/internal/telephony/MccTable.java b/frameworks/opt/telephony/src/java/com/android/internal/telephony/MccTable.java
index c00b4a4..d0b80f8 100644
--- a/frameworks/opt/telephony/src/java/com/android/internal/telephony/MccTable.java
+++ b/frameworks/opt/telephony/src/java/com/android/internal/telephony/MccTable.java
@@ -204,37 +204,6 @@ public final class MccTable
                 setTimezoneFromMccIfNeeded(context, mcc);
                 locale = getLocaleFromMcc(context, mcc);
             }
-            if (fromServiceState) {
-                setWifiCountryCodeFromMcc(context, mcc);
-            } else {
-                // from SIM
-                try {
-                    Configuration config = new Configuration();
-                    boolean updateConfig = false;
-                    if (mcc != 0) {
-                        config.mcc = mcc;
-                        config.mnc = mnc == 0 ? Configuration.MNC_ZERO : mnc;
-                        updateConfig = true;
-                    }
-                    if (locale != null) {
-                        config.setLocale(locale);
-                        updateConfig = true;
-                    }
-                    if (updateConfig) {
-                        Slog.d(LOG_TAG, "updateMccMncConfiguration updateConfig config=" + config);
-                        ActivityManagerNative.getDefault().updateConfiguration(config);
-                    } else {
-                        Slog.d(LOG_TAG, "updateMccMncConfiguration nothing to update");
-                    }
-                } catch (RemoteException e) {
-                    Slog.e(LOG_TAG, "Can't update configuration", e);
-                }
-            }
-        } else {
-            if (fromServiceState) {
-                // an empty mccmnc means no signal - tell wifi we don't know
-                setWifiCountryCodeFromMcc(context, 0);
-            }
         }
     }
 
@@ -364,21 +333,6 @@ public final class MccTable
         return getLocaleForLanguageCountry(context, language, country);
     }
 
-    /**
-     * Set the country code for wifi.  This sets allowed wifi channels based on the
-     * country of the carrier we see.  If we can't see any, reset to 0 so we don't
-     * broadcast on forbidden channels.
-     * @param context Context to act on.
-     * @param mcc Mobile Country Code of the operator.  0 if not known
-     */
-    private static void setWifiCountryCodeFromMcc(Context context, int mcc) {
-        String country = MccTable.countryCodeForMcc(mcc);
-        Slog.d(LOG_TAG, "WIFI_COUNTRY_CODE set to " + country);
-        WifiManager wM = (WifiManager) context.getSystemService(Context.WIFI_SERVICE);
-        //persist
-        wM.setCountryCode(country, true);
-    }
-
     static {
         sTable = new ArrayList<MccEntry>(240);
 
